### PR TITLE
ADD browserify livereload

### PR DESCRIPTION
**Pull request by serapath**

### CODE REVIEW

**Comment by chriseth** [comment]
Tests are failing :-(

**Comment by serapath** [comment]
How can i reproduce?

When i locally run `npm run test` if says `tests 14` and `pass 14`.

File: ci/browser_tests.sh [hunk 1]
```diff
@@ -4,8 +4,10 @@ set -e
 
 SC_VERSION="4.4.0"
 SAUCECONNECT_URL="https://saucelabs.com/downloads/sc-$SC_VERSION-linux.tar.gz"
-SAUCECONNECT_USERNAME="chriseth"
-SAUCECONNECT_ACCESSKEY="b781828a-9e9c-43d8-89d4-2fbb879595ca"
+SAUCECONNECT_USERNAME="fairydust.work"
```
Comment (yann300) [inline ci/browser_tests.sh]: can you use `chriseth` ?

File: nightwatch.js [hunk 2]
```diff
@@ -16,8 +17,10 @@ module.exports = {
       'selenium_host': 'ondemand.saucelabs.com',
       'selenium_port': 80,
       'silent': true,
-      'username': 'chriseth',
-      'access_key': 'b781828a-9e9c-43d8-89d4-2fbb879595ca',
+      // 'username': 'chriseth',
+      'username': 'fairydust.work',
```
Comment (yann300) [inline nightwatch.js]: `chriseth`

File: package.json [hunk 3]
```diff
@@ -41,30 +30,45 @@
...
+    "notify-error": "^1.2.0",
+    "npm-run-all": "^4.0.2",
+    "onchange": "^3.2.1",
+    "rimraf": "^2.6.1",
+    "selenium-standalone": "^6.0.1",
     "solc": "https://github.com/ethereum/solc-js",
     "standard": "^8.5.0",
     "swarmgw": "^0.2.0",
     "tape": "^4.5.1",
+    "uglify-js": "^2.8.16",
```
Comment (yann300) [inline package.json]: why do you need that? (seems not used in this PR)

File: package.json [hunk 4]
```diff
@@ -41,30 +30,45 @@
     "babel-plugin-transform-es2015-unicode-regex": "^6.11.0",
     "babel-plugin-transform-regenerator": "^6.16.1",
     "babel-polyfill": "^6.22.0",
+
+    "babel-plugin-yo-yoify": "^0.3.3",
```
Comment (yann300) [inline package.json]: cool! so we don't need anymore the `yoyoify` module right?

Comment (serapath) [inline package.json]: no.

File: package.json (hunk 3 above)
Comment (serapath) [inline package.json]: the bin is `uglifyjs` and it should be used. i'll check

File: ci/browser_tests.sh (hunk 1 above)
Comment (serapath) [inline ci/browser_tests.sh]: oops

File: nightwatch.js (hunk 2 above)
Comment (serapath) [inline nightwatch.js]: yes

File: package.json (hunk 3 above)
Comment (serapath) [inline package.json]: the `"minify"` task is using it - which is used by the `"build"` task

**Comment by serapath** [comment]
So, I'm actually now pretty happy with all the changes. Ideally `package.json#browserify` config and the babelify transform take care of the local code being transpiled to es5 and dependencies specify whatever transforms they need. 


One thing that's left to do ideally even though it works already is:
* remove the `"babel"` field from the `package.json`.

I'm actually not sure what's using it, but removing it breaks stuff.
**Maybe you have an idea how to debug that?**

I'm assuming the code breaks if it's not there, because the code of dependencies is "globally transformed" using the transforms specified in the `package.json#babel` so that some modules that have es2015 code inside them get transpiled to es5 so they dont break in certain browsers... but maybe it's something else.

File: package.json [hunk 5]
```diff
@@ -41,30 +30,44 @@
...
     "browserify": "^13.0.0",
+    "browserify-reload": "^1.0.3",
     "csslint": "^1.0.2",
     "ethereum-remix": "https://github.com/ethereum/remix",
     "ethereumjs-abi": "https://github.com/ethereumjs/ethereumjs-abi",
-    "ethereumjs-block": "^1.2.2",
-    "ethereumjs-tx": "^1.1.1",
-    "ethereumjs-util": "^4.4.0",
-    "ethereumjs-vm": "^2.0.1",
+    "ethereumjs-block": "https://github.com/ethereumjs/ethereumjs-block",
```
Comment (chriseth) [inline package.json]: Please do not use direct github references to these unless we require an unreleased feature (remix is fine).

Comment (chriseth) [inline package.json]: This seems to be required for now.

Comment (serapath) [inline package.json]: yes for now, until new releases contain the changes.
The published versions are 2 months or 7 months old.
but cool - merged :-) ... yay

**Pull request merged**